tools/flash_shelly.py

```diff
@@ -266,16 +266,16 @@ def write_flash(host, lfw, dlurl, cfw_type, mode):
     try:
       with urllib.request.urlopen(checkurl) as fp:
         info = json.load(fp)
+      if mode == 'homekit':
+        onlinecheck = info['version']
+      else:
+        onlinecheck = info['fw'].split('/v')[1].split('@')[0]
```

Comment:
note that this format only applies to stock fw 1.8 and later, previous versions do not follow this pattern and will not work properly.

Reply:
yeah, but you say, people should update to newest stock firmware for trying to flash HomeKit, so imo it fine ?

Reply:
suposed i could add a fall back to just info['fw']

Reply:
well, it's ok to let them know that politely, but have script blow up is not good.

btw, can we make the script ensure that device is running latest stock before flashing homekit?
it is not yet a strict requirement but will be in the future.

Reply:
if it fails to get correct version info it now reports as `0.0.0`

if device has stock firmware on it, it now check if it is latest, if it is not it reports this, this also includes a device that has really old firmware on that could not be read

this is only for trying to flash to HomeKit 
```
Probing Shelly device for info...

Host: shellydimmer2-40F5200092B2.local
Model: Shelly1
Current: Official 0.0.0
Latest: Official 1.8.5
This device needs to be upgraded to latest stock firmware version, before you can flash to HomeKit (y/n) ? n
Skipping Flash...
```

if using `-m keep`
```
Probing Shelly device for info...

Host: shellydimmer2-40F5200092B2.local
Model: Shelly1
Current: Official 0.0.0
Latest: Official 1.8.5
Would have been upgraded to version 1.8.5 (y/n) ? n
Skipping Flash...
```

Reply:
```
Host: shellydimmer2-40F5200092B2.local
Model: ShellyDimmer2
Current: Official 1.8.0
Latest: Official 1.8.5
D_URL: http://api.shelly.cloud/firmware/SHDM-2.zip
perform_flash: True
This device needs to be upgraded to latest stock firmware version, before you can flash to HomeKit (y/n) ? y   

write_flash
Now Flashing...
Successfully flashed shellydimmer2-40F5200092B2 to 1.8.5
 
Andrews-MacBook-Pro:tools andrewblackburn$ ./flash_shelly.py shellydimmer2-40F5200092B2
Probing Shelly device for info...

Host: shellydimmer2-40F5200092B2.local
Model: ShellyDimmer2
Current: Official 1.8.5
Latest: HomeKit Not available
Is not supported yet...

```